src/builder/build_traits.rs

```diff
@@ -612,6 +612,14 @@ pub trait Dataflow: Container {
                 })
             }
         };
+        let signature = type_scheme.instantiate(type_args, exts).map_err(|e| {
+            BuildError::InvalidHUGR(ValidationError::SignatureError {
+                // TODO this is rather a horrendous hack. Do we need some way of returning a SignatureError without a node
```

Comment:
I guess we should make a new error constructor for this

Reply:
Yes, done, hope the comment clarifies